test/puppetlabs/puppetdb/integration/resource_events.clj

```diff
@@ -0,0 +1,46 @@
+(ns puppetlabs.puppetdb.integration.resource-events
+  (:require [clojure.test :refer :all]
+            [me.raynes.fs :as fs]
+            [puppetlabs.puppetdb.integration.fixtures :as int]
+            [puppetlabs.puppetdb.testutils.services :as svc-utils]
+            [clj-time.core :refer [now]]
+            [clj-time.core :refer [now days plus]]
+            [metrics.counters :as counters]
+            [puppetlabs.puppetdb.cli.services :as pdb-services]
+            [puppetlabs.puppetdb.testutils :as tu]))
+
+(defn read-gc-count-metric []
+  ;; metrics are global, so...
+  (counters/value (:resource-events-purges pdb-services/admin-metrics)))
+
+(deftest ^:integration resource-events-ttl
+  (with-open [pg (int/setup-postgres)]
+    (with-open [pdb (int/run-puppetdb pg {})
+                ps (int/run-puppet-server [pdb] {})]
+      (testing "Run agent once to populate database"
+        (int/run-puppet-as "ttl-agent" ps pdb "notify { 'irrelevant manifest': }"))
+
+      (testing "Verify we have resource events"
+        (is (= 1 (count (int/pql-query pdb "events { timestamp > 0 }"))))))
+
+    (testing "Sleep for one second to make sure we have a ttl to exceed"
+      (Thread/sleep 1000))
```

Comment:
When adding another recent test, I found out that we can probably set `ms` ttls, i.e. `1ms`.  Might want to double-check me on that (and I'll need to fix the other code if I was wrong), but if that's right, then might be worth changing this sleep to say 2ms or something.  Doesn't matter for this one test, but over time, they add up...

Reply:
Ahhh, I'll give that a shot here